src/Method/QueryParameter.php

```diff
@@ -17,17 +19,17 @@ final class QueryParameter implements AuthenticationMethodInterface
 {
     private string $parameterName = 'access-token';
 
-    private IdentityRepositoryInterface $identityRepository;
+    private IdentityWithTokenRepositoryInterface $identityRepository;
 
-    public function __construct(IdentityRepositoryInterface $identityRepository)
+    public function __construct(IdentityWithTokenRepositoryInterface $identityRepository)
     {
         $this->identityRepository = $identityRepository;
     }
 
     public function authenticate(ServerRequestInterface $request): ?IdentityInterface
     {
         $accessToken = $request->getQueryParams()[$this->parameterName] ?? null;
-        if (\is_string($accessToken)) {
+        if (is_string($accessToken)) {
             return $this->identityRepository->findIdentityByToken($accessToken, self::class);
```

Comment:
I think second parameter should be configurable like the `parameterName` property.
Same for `HttpHeader:45` and `HttpBasic:55`